ParallelPrimitives/RadixSort.cpp

```diff
@@ -170,7 +170,16 @@ void RadixSort::compileKernels( const std::string& kernelPath, const std::string
 			const auto sort_num_warps_param = "-DSORT_NUM_WARPS_PER_BLOCK_VAL=" + std::to_string( m_num_warps_per_block_for_sort );
 
 			std::vector<const char*> opts;
-			opts.push_back( "-ffast-math" );
+
+			if( const std::string device_name = m_props.name; device_name.find( "AMD" ) != std::string::npos )
```

Comment:
It's better to test `NVIDIA` instead since AMD driver are not reliable (especially internal ones) and sometimes doesn't return correct name.